File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefix=[]
        c=0
        for i in nums:
            c+=i
            self.prefix.append(c)

    def sumRange(self, left: int, right: int) -> int:
        rSum=self.prefix[right]
        lSum=self.prefix[left-1] if left>0 else 0
        return rSum-lSum


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {
    private ArrayList<Integer> prefix;
    public NumArray(int[] nums) {
        prefix= new ArrayList<>();
        int c=0;
        for(int i:nums){
            c+=i;
            prefix.add(c);
        }
    }
    
    public int sumRange(int left, int right) {
        int rSum=prefix.get(right);
        int lSum= (left>0)? prefix.get(left-1): 0;
        return rSum-lSum;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */